### PR TITLE
Adapt the code to v0.12.x

### PR DESCRIPTION
**Pull request by ainestal**
Some code changes so the module can be used with the new terraform v0.12.x version.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-  default = []
+  type        = list(string)
+  default     = []
 }
 
 variable "allow_password_change_without_mfa" {
   description = "Allow changing the user password without MFA"
-  type = "string"
-  default = "false"
+  type        = string
```
Comment (jeromegamez) [inline variables.tf]: Could we use the occasion to make this a `bool` here (and in the README respectively)?

Comment (ainestal) [reply variables.tf]: sure, let me do it and test it locally. I'll send another commit

**Comment by jeromegamez** [comment]
@ainestal Thanks for the additional commits, I didn't get notified, so sorry for the late reply. Could you point me to the docs where it the `&{...}` notation is described instead of using `$${...}`?

I can't really test the changes at the moment, so I'm asking a lot of questions to make as sure as possible that I don't oversee something 😅

**Comment by jeromegamez** [comment]
While looking through the AWS docs and the policies I saw that https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html uses the `${...}`s, and I used `$${...}` to escape the `$` sign... When using `&{...}`, wouldn't this break then?

**Comment by ainestal** [comment]
@jeromegamez you can see the `&{...}` notation here https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html#context-variable-interpolation

I tested it and it works fine :)

**Comment by jeromegamez** [comment]
Thank you for the link, I assumed that you got it working already, just wanted to be sure! I'll merge this and create a 2.0 release so that we can concentrate on #2 🎉

**Pull request merged**